scripts/kind-e2e/e2e.sh

```diff
@@ -52,52 +71,46 @@ function kind_fixup_config() {
     fi
 }
 
-function kind_clusters() {
-    status=$1
-    version=$2
-    pids=(-1 -1 -1)
-    logs=()
-    for i in 1 2 3; do
-        export KUBECONFIG=${PRJ_ROOT}/output/kind-config/dapper/kind-config-cluster${i}
-        if [[ $(kind get clusters | grep cluster${i} | wc -l) -gt 0  ]]; then
-            echo Cluster cluster${i} already exists, skipping cluster creation...
-            kind export kubeconfig --name=cluster${i}
-            kind_fixup_config cluster${i}
-        else
-            logs[$i]=$(mktemp)
-            echo Creating cluster${i}, logging to ${logs[$i]}...
-            (
-            generate_cluster_yaml "cluster${i}"
-            if [[ -n ${version} ]]; then
-                kind create cluster --image=kindest/node:v${version} --name=cluster${i} --config=${PRJ_ROOT}/scripts/kind-e2e/cluster${i}-config.yaml
-            else
-                kind create cluster --name=cluster${i} --config=${PRJ_ROOT}/scripts/kind-e2e/cluster${i}-config.yaml
-            fi
-            kind_fixup_config cluster${i}
-            ) > ${logs[$i]} 2>&1 &
-            set pids[$i] = $!
-        fi
-    done
-    print_logs "${logs[@]}"
+function create_kind_cluster() {
+    export KUBECONFIG=${DAPPER_SOURCE}/output/kind-config/dapper/kind-config-${cluster}
+    if [[ $(kind get clusters | grep "^${cluster}$" | wc -l) -gt 0  ]]; then
+        echo "KIND cluster ${cluster} already exists, skipping it's creation..."
+        kind export kubeconfig --name=${cluster}
+        kind_fixup_config ${cluster}
+        return
+    fi
+
+    echo Creating KIND cluster: ${cluster}...
+    generate_cluster_yaml "${cluster}"
+    image_flag=''
+    if [[ -n ${version} ]]; then
+        image_flag="--image=kindest/node:v${version}"
+    fi
+
+    kind create cluster $image_flag --name=${cluster} --config=${DAPPER_SOURCE}/scripts/kind-e2e/${cluster}-config.yaml
+    kind_fixup_config ${cluster}
 }
 
-function setup_custom_cni(){
-    for i in 2 3; do
-        if kubectl --context=cluster${i} wait --for=condition=Ready pods -l name=weave-net -n kube-system --timeout=60s > /dev/null 2>&1; then
-            echo "Weave already deployed cluster${i}."
-        else
-            echo "Applying weave network in to cluster${i}..."
-            kubectl --context=cluster${i} apply -f "https://cloud.weave.works/k8s/net?k8s-version=$(kubectl version | base64 | tr -d '\n')&env.IPALLOC_RANGE=${cluster_CIDRs[cluster${i}]}"
-            echo "Waiting for weave-net pods to be ready cluster${i}..."
-            kubectl --context=cluster${i} wait --for=condition=Ready pods -l name=weave-net -n kube-system --timeout=300s
-            echo "Waiting for core-dns deployment to be ready cluster${i}..."
-            kubectl --context=cluster${i} -n kube-system rollout status deploy/coredns --timeout=300s
-        fi
-    done
+function use_kube_context() {
+    kubectl config use-context $1
```

Comment:
Could we avoid setting the context globally? It makes it harder to do anything in parallel with an ongoing deployment.

Reply:
I'm working on getting rid of it